openlcs/libs/kojiconnector.py

```diff
@@ -325,5 +329,72 @@ def download_container_image_archives(self, build, dest_dir, arch=None):
                 for archive in archives:
                     self.download_archive(build, type_path, archive, dest_dir)
             else:
-                err_msg = "No images found."
-                raise ValueError(err_msg)
+                raise ValueError("No images found.") from None
+
+    @staticmethod
+    def get_remote_source_component_flat(data):
+        return {
+            "uuid": str(uuid.uuid4()),
+            "type": data.get("type").upper(),
+            "name": data.get("name"),
+            "version": data.get("version", ""),
+            "release": "",
+            "license": "",
+            "arch": "",
+            'synced': False
+        }
+
+    def parse_remote_source_components(self, rs_archive_url):
+        """
+        Parse components from remote source archive url.
```

Comment:
I guess the rs_archive_url should be the binary container's remote-source.json, not the remote-source.tar.gz. So would you like to give more description?

Reply:
Please check the function `get_remote_source_components` in `kojiconnector.py`, this `rs_archive_url ` is the url for `remote-source.json`